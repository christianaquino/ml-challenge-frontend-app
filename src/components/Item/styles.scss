@use 'sass:map';
@use '../../global.scss' as *;

.item {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  border-bottom: 1px solid map.get($colors, 'ml-gray-1');
  padding: 16px 0;
  grid-template-areas:
    'image price price ... freeshipping '
    'image title title title ...'
    'image condition ... ... ...'
    'image ... ... ... ...';

  &--image {
    height: 180px;
    max-width: 180px;
  }

  &--imagecontainer {
    grid-area: image;
    width: 180px;
    padding-right: 12px;
    border-radius: 4px;
  }

  &--price {
    grid-area: price;
    font-size: 24px;
    padding-top: 16px;
    padding-bottom: 32px;
  }

  &--title {
    grid-area: title;
    font-size: 18px;
    color: map.get($colors, 'ml-gray-4');
  }

  &--condition {
    grid-area: condition;
    color: map.get($colors, 'ml-gray-4');
  }

  &--shipping-free {
    grid-area: freeshipping;
    font-size: 12px;
    padding-top: 12px;
    color: map.get($colors, 'ml-gray-3');
  }
}
