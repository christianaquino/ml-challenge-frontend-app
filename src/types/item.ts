export interface IPrice {
  currency: string,
  amount: Number
}

export interface IItem {
  id: string,
  picture: string,
  title: string,
  price: IPrice
  condition: string
  free_shipping: boolean
}

export interface IItemDetails extends IItem {
  sold_quantity: Number,
  description: string
}
