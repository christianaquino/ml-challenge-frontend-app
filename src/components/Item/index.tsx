import React from 'react';
import './styles.scss';

function Item({ item }: any) {
  return (
    <div className="item">
      <div className="item--imagecontainer">
        <img src={item.picture} alt={item.title} className="item--image" />
      </div>
      <div className="item--price">{`${item.price.currency} ${item.price.amount}`}</div>
      <div className="item--title">{item.title}</div>
      <div className="item--condition">{item.condition}</div>
      { item.free_shipping && <div className="item--shipping-free">Env√≠o Gratis</div>}
    </div>
  );
}

export default Item;
