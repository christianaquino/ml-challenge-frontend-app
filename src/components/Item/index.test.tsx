import { render } from '@testing-library/react';
import { ClassAttributes, AnchorHTMLAttributes } from 'react';
import Item from '.';
import { IItem } from '../../types/item';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  Link: (
    // eslint-disable-next-line no-undef
    props: JSX.IntrinsicAttributes
      & ClassAttributes<HTMLAnchorElement>
      & AnchorHTMLAttributes<HTMLAnchorElement>,
  ) => (
    // eslint-disable-next-line react/jsx-props-no-spreading
    <a {...props}>
      { // eslint-disable-next-line react/destructuring-assignment
          props.children
        }
    </a>
  ),
}));

describe('Item component', () => {
  const item: IItem = {
    id: 'MLC8752432',
    picture: '//item.jpg',
    title: 'IPhone 12',
    price: {
      amount: 650000,
      currency: 'CLP',
    },
    condition: 'Nuevo',
    free_shipping: true,
  };
  test('renders correctly itme with free shipping', () => {
    const { asFragment } = render(<Item item={item} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly itme with no free shipping', () => {
    const { asFragment } = render(<Item item={{ ...item, free_shipping: false }} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
