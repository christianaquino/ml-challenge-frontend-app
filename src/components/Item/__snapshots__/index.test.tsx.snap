// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders correctly itme with free shipping 1`] = `
<DocumentFragment>
  <a
    class="item"
    to="/items/MLC8752432"
  >
    <div
      class="item--imagecontainer"
    >
      <img
        alt="IPhone 12"
        class="item--image"
        src="//item.jpg"
      />
    </div>
    <div
      class="item--price"
    >
      CLP 650000
    </div>
    <div
      class="item--title"
    >
      IPhone 12
    </div>
    <div
      class="item--condition"
    >
      Nuevo
    </div>
    <div
      class="item--shipping-free"
    >
      Env√≠o Gratis
    </div>
  </a>
</DocumentFragment>
`;

exports[`Item component renders correctly itme with no free shipping 1`] = `
<DocumentFragment>
  <a
    class="item"
    to="/items/MLC8752432"
  >
    <div
      class="item--imagecontainer"
    >
      <img
        alt="IPhone 12"
        class="item--image"
        src="//item.jpg"
      />
    </div>
    <div
      class="item--price"
    >
      CLP 650000
    </div>
    <div
      class="item--title"
    >
      IPhone 12
    </div>
    <div
      class="item--condition"
    >
      Nuevo
    </div>
  </a>
</DocumentFragment>
`;
